cdap-api/src/main/java/co/cask/cdap/api/workflow/Workflow.java

```diff
@@ -15,13 +15,15 @@
  */
 package co.cask.cdap.api.workflow;
 
+import co.cask.cdap.api.ProgramLifecycle;
+
 /**
  * Defines an interface for the Workflow.
  */
-public interface Workflow {
+public interface Workflow extends ProgramLifecycle<WorkflowContext> {
```

Comment:
This change is not backward compatible. You can't expect everyone is extending from AbstractWorkflow